apps/portal/js/designer.js

```diff
@@ -1835,69 +1835,49 @@ $(function () {
 
-                if (recentlyOpenedPageProperty == pid) {
-                    recentlyOpenedPageProperty = null;
-                }
-
-                updatePagesList(pages);
-                initNanoScroller();
-                if (!pages.length) {
-                    dashboard.landing = null;
-                    hideProperties();
-                    initFirstPage();
-                    return;
-                }
-
-                var first = pages[0];
-                if (pid === dashboard.landing) {
-                    dashboard.landing = first.id;
-                }
+                    if (recentlyOpenedPageProperty == pid) {
+                        recentlyOpenedPageProperty = null;
+                    }
 
-                saveDashboard();
-                if (pid !== page.id) {
                     updatePagesList(pages);
                     initNanoScroller();
-                    return;
-                }
-                hideProperties();
-                renderPage(first.id);
-            });
-        });
-        pagesMenu.find(".ues-pages-list").on('click', 'li a .ues-trash', function (e) {
-            e.stopPropagation();
-            var thiz = $(this);
-            var pid = thiz.parent().data('id');
-            removePage(pid, DEFAULT_DASHBOARD_VIEW, function (err) {
-                var pages = dashboard.pages;
-                updatePagesList(pages);
-                initNanoScroller();
-                if (!pages.length) {
-                    dashboard.landing = null;
-                    hideProperties();
-                    initFirstPage();
-                    return;
-                }
+                    if (!pages.length) {
+                        dashboard.landing = null;
+                        hideProperties();
+                        initFirstPage();
+                        return;
+                    }
 
-                var first = pages[0];
-                if (pid === dashboard.landing) {
-                    dashboard.landing = first.id;
-                }
+                    var first = pages[0];
+                    if (pid === dashboard.landing) {
+                        dashboard.landing = first.id;
+                    }
 
-                saveDashboard();
-                if (pid !== page.id) {
-                    updatePagesList(pages);
-                    initNanoScroller();
-                    return;
-                }
-                hideProperties();
-                renderPage(first.id);
-            });
+                    saveDashboard();
+                    if (pid !== page.id) {
+                        updatePagesList(pages);
+                        initNanoScroller();
+                        return;
+                    }
+                    hideProperties();
+                    renderPage(first.id);
+                });
+            };
+
+            generateMessage("This will remove the page and all its content." +
```

Comment:
AFAIK you can define i18 variable in global level(in this case parent jaggery file ) and use in js file